src/nvim/misc1.c

```diff
@@ -2691,6 +2679,41 @@ void fast_breakcheck(void)
   }
 }
 
+// Call shell. Calls os_call_shell, with 'shellxquote' added.
+int call_shell(char_u *cmd, ShellOpts opts, char_u *extra_shell_arg)
```

Comment:
Moving this to misc1.c (from misc2.c) to avoid too much churn in this PR.

In another PR we might want to introduce a `core.c` module for things like exiting, startup, shell calls, and other general application facilities.